sys/net/wot/coap/coap.c

```diff
@@ -105,6 +98,51 @@ void _wot_td_coap_ser_receiver(const char *c){
     _wot_td_coap_plen += coap_blockwise_put_char(&_wot_td_coap_slicer, _wot_td_coap_buf+_wot_td_coap_plen, (char) *c);
 }
 
+static int get_base_ip_address(ipv6_addr_t *res) {
+    const int MAX_ADRESSES_TO_CHECK = 5;
+    netif_t* interface = NULL;
+    ipv6_addr_t local_address = {0};
+    ipv6_addr_t ula_address = {0};
+    bool link_local_found = false;
+    bool ula_found = false;
+    int netres;
+
+    while ((interface = netif_iter(interface)) != NULL) {
```

Comment:
@Citrullin Do you think only the net interface that received the request should be checked for ip adresses? I think that should be relatively easy to implement (using `sock_aux_local`).

Reply:
I merged the PR now into the `wot_gcoap` branch but this is probably something we should keep in mind.